src/main/java/com/google/sps/ProductDetectionAPI.java

```diff
@@ -0,0 +1,24 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import com.google.sps.data.ProductDetectionData;
+
+/**
+ * Interface for ProductDetectionAPIImpl, respective FakeProductDetectionAPI, used for testing.
+ */
+public interface ProductDetectionAPI {
```

Comment:
I was thinking to keep it `protected` as we have all the files we use this in, are also in the same `sps` package.
```suggestion
protected interface ProductDetectionAPI {
```
And in the next line too.

Reply:
Actually, that class is a top level class, therefore it cannot be `protected`, as stated, for instance, [here](https://www.tutorialspoint.com/can-we-declare-a-top-level-class-as-protected-or-private-in-java). Also, [here](https://stackoverflow.com/questions/3869556/why-can-a-class-not-be-defined-as-protected) is an explanation for that.